packages/xo-web/src/common/xo/index.js

```diff
@@ -3220,15 +3220,36 @@ export const upgradeProxyAppliance = proxy =>
 export const getProxyApplianceUpdaterState = id =>
   _call('proxy.getApplianceUpdaterState', { id })
 
-export const checkProxyHealth = proxy =>
-  _call('proxy.checkHealth', { id: resolveId(proxy) }).then(() =>
-    success(
-      <span>
-        <Icon icon='success' /> {_('proxyTestSuccess', { name: proxy.name })}
-      </span>,
-      _('proxyTestSuccessMessage')
-    )
-  )
+const PROXY_HEALTH_CHECK_COMMON_ERRORS_CODE = new Set([
+  'ECONNREFUSED',
+  'ECONNRESET',
+  'EHOSTUNREACH',
+  'ENOTFOUND',
+  'ETIMEDOUT',
+])
+
+export const checkProxyHealth = async proxy => {
+  const result = await _call('proxy.checkHealth', {
+    id: resolveId(proxy),
+  })
```

Comment:
use `.then` and you don't need to update `checkHealth` in `xo-mixins/proxies.js`.

Reply:
`checkProxyHealth` should not throw an error to say that the proxy has an issue because:
1. we'll don't know if the error is due to the connection between the proxy and the XOA or it's due to a `checkProxyHealth` implementation error.
2. the error isn't a JSON RPC error so it will be an `unknown error from a peer`.